flycheck.el

```diff
@@ -6082,6 +6082,18 @@ objects)."
 
 
 ;;; Convenience definition of command-syntax checkers
+(eval-and-compile
+  ;; We need the byte-compiler to be aware of this function when compiling
+  ;; `flycheck-define-checker'.
```

Comment:
Technically, when macroexpanding flycheck-define-checker.
I think it'd also be good to document why this function is a problem (namely, the byte-compiler needs to be aware of it, but when-eval-after-load can run before that.